src/net_processing.cpp

```diff
@@ -1779,22 +1781,23 @@ std::optional<std::string> PeerManagerImpl::FetchBlock(NodeId peer_id, const CBl
 }
 
 std::unique_ptr<PeerManager> PeerManager::make(CConnman& connman, AddrMan& addrman,
-                                               BanMan* banman, ChainstateManager& chainman,
-                                               CTxMemPool& pool, bool ignore_incoming_txs)
+                                               ChainstateManager& chainman, CTxMemPool& pool,
+                                               Options opts)
 {
-    return std::make_unique<PeerManagerImpl>(connman, addrman, banman, chainman, pool, ignore_incoming_txs);
+    return std::make_unique<PeerManagerImpl>(connman, addrman, chainman, pool, opts);
 }
 
 PeerManagerImpl::PeerManagerImpl(CConnman& connman, AddrMan& addrman,
-                                 BanMan* banman, ChainstateManager& chainman,
-                                 CTxMemPool& pool, bool ignore_incoming_txs)
+                                 ChainstateManager& chainman, CTxMemPool& pool,
+                                 Options opts)
     : m_chainparams(chainman.GetParams()),
       m_connman(connman),
       m_addrman(addrman),
-      m_banman(banman),
+      m_banman(opts.banman),
```

Comment:
Doesn't this leave us with two pointers, one in `m_banman` and the other in `m_opts.banman` that is never used?

Reply:
Yea, could get rid of m_banman or not have banman in the options in the first place...

Reply:
Will leave as is, can be done in a follow up

Reply:
As discussed offline, what I meant with "leave it as is" is to not move banman in this PR, i.e. reverting the change in this PR.

Reply:
Removed banman from the options on latest push